### PR TITLE
fix: sync the remapping of MaterialSwitchClip property names 

### PR DESCRIPTION
**Pull request by smnwttbr**

### CODE REVIEW

File: Editor/MaterialSwitchClipEditor.cs [hunk 1]
```diff
@@ -304,13 +305,15 @@ private void ShowRemoveOverrideButton(SerializedProperty itemProperty)
...
-            string arrayPropertyPath, System.Action<SerializedProperty> guiMethod = null)
+        private void DrawPropertyOverrideList(SerializedProperty ppm, string togglePropertyPath, string heading, string arrayPropertyPath, System.Action<SerializedProperty, string> guiMethod = null)
         {
             var showPropertyListToggle = ppm.FindPropertyRelative(togglePropertyPath);
             showPropertyListToggle.boolValue = EditorGUILayout.Foldout(showPropertyListToggle.boolValue, heading);
+            
             if (showPropertyListToggle.boolValue)
             {
+                var materialProperty = ppm.FindPropertyRelative(nameof(MaterialProperties.material));
+                var material = materialProperty.objectReferenceValue as Material;
```
Comment (sindharta) [inline Editor/MaterialSwitchClipEditor.cs:329]: what happens if material is null here ?

File: Editor/MaterialSwitchClipEditor.cs [hunk 2]
```diff
@@ -185,11 +185,14 @@ private void DrawPalettePropertyMapUI(SerializedProperty ppm, SerializedProperty
         {
             GUILayout.BeginVertical("box");
             EditorGUI.indentLevel--;
+            SerializedProperty materialProperty = ppm.FindPropertyRelative(nameof(MaterialProperties.material));;
+            var material = materialProperty.objectReferenceValue as Material;
```
Comment (sindharta) [inline Editor/MaterialSwitchClipEditor.cs]: This `material` variable doesn't seem to be used ?

File: Editor/MaterialSwitchEditorUtility.cs [hunk 3]
```diff
@@ -14,6 +14,9 @@ namespace Unity.MaterialSwitch {
 /// Utility class to perform various functions on MaterialSwitch classes.
 /// </summary>
 public static class MaterialSwitchEditorUtility {
+    
+    private static Dictionary<Shader,MaterialPropertyNameMap> _nameRemaps;
```
Comment (sindharta) [inline Editor/MaterialSwitchEditorUtility.cs]: This is static, and not necessarily refreshed when we create a new MaterialPropertyNameMap asset

**Comment by sindharta** [comment]
Please also fix the conflict

File: Editor/MaterialSwitchClipEditor.cs (hunk 1 above)
Comment (smnwttbr) [reply Editor/MaterialSwitchClipEditor.cs:329]: If the material is null, the propertyName is used as the display name.

File: Editor/MaterialSwitchClipEditor.cs (hunk 2 above)
Comment (sindharta) [reply Editor/MaterialSwitchClipEditor.cs]: @simonwittber I think we can remove this line

File: Editor/MaterialSwitchEditorUtility.cs (hunk 3 above)
Comment (sindharta) [reply Editor/MaterialSwitchEditorUtility.cs]: @simonwittber I think you haven't addressed the above comment ?

**Review by sindharta — COMMENTED** [review]
Please fix the merge conflicts too

**Comment by simonwittber** [comment]
Correct, this fix is in progress.

    :: Simon Wittber

File: Editor/RemapNameCache.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace Unity.MaterialSwitch
+{
+    public class RemapNameCache
```
Comment (sindharta) [inline Editor/RemapNameCache.cs]: please make this to internal

File: Editor/RemapNameCache.cs [hunk 5]
```diff
@@ -16,12 +16,15 @@ internal class RemapNameCache
...
                 {
-                    if (nameMap.TryGetValue(propertyName, out var remappedName))
+                    if(map.shader == material.shader)
                     {
-                        displayName = remappedName.displayName;
-                        isHidden = remappedName.hidden;
+                        if (map.TryGetValue(propertyName, out var remappedName))
+                        {
+                            displayName = remappedName.displayName;
+                            isHidden = remappedName.hidden;
```
Comment (sindharta) [inline Editor/RemapNameCache.cs:26]: can we do early return here ?

**Pull request merged**